ember-core/shared/src/main/scala/org/http4s/ember/core/Parser.scala

```diff
@@ -512,11 +511,18 @@ private[ember] object Parser {
     }
   }
 
-  private def combineArrays[A: scala.reflect.ClassTag](a1: Array[A], a2: Array[A]): Array[A] = {
-    val buff = mutable.ArrayBuffer[A]()
-    buff.++=(a1)
-    buff.++=(a2)
-    buff.toArray
+  private[this] def appendByteArrays(a1: Array[Byte], a2: Chunk[Byte]): Array[Byte] = {
+    val res = new Array[Byte](a1.length + a2.size)
+    var ix = 0
+    while (ix < a1.length) {
+      res(ix) = a1(ix)
+      ix += 1
+    }
+    while (ix < a2.size) {
+      res(ix) = a2(ix)
+      ix += 1
+    }
```

Comment:
Is the math here right, shouldn't indices for `a2` start again from zero?

Reply:
Thanks! Fixed.